app/(app)/(purchase-and-payment)/purchase/choose-vehicle.tsx

```diff
@@ -1,51 +1,109 @@
-
   const { licencePlate } = usePurchaseStoreContext()
-  const onPurchaseStoreUpdate = usePurchaseStoreUpdateContext()
+  const { vehicles, isVehiclePresent, getVehicle } = useVehicles()
+
+  const [oneTimeLicencePlate, setOneTimeLicencePlate] = useState(
+    getVehicle(licencePlate)?.oneTimeUse ? licencePlate : '',
+  )
+  const [oneTimeLicencePlateError, setOneTimeLicencePlateError] = useState('')
 
-  const { vehicles } = useVehicles()
+  const onPurchaseStoreUpdate = usePurchaseStoreUpdateContext()
 
   const handleChoseVehicle = (newLicencePlate: string) => {
     onPurchaseStoreUpdate({ licencePlate: newLicencePlate })
     router.push('/purchase')
   }
 
+  const handleLicencePlateChange = (newLicencePlate: string) => {
+    setOneTimeLicencePlate(newLicencePlate)
+
+    const newSanitizedLicencePlate = sanitizeLicencePlate(newLicencePlate)
+
+    setOneTimeLicencePlateError('')
+    if (isVehiclePresent(newSanitizedLicencePlate)) {
+      setOneTimeLicencePlateError(t('licencePlateDuplicate'))
+    }
+  }
+
   return (
-    <ScreenView title={t('title')}>
+    <ScreenView>
+      <Stack.Screen
+        options={{
+          title: t('title'),
+          headerRight: () => (
+            <Button
+              variant="plain"
+              disabled={!oneTimeLicencePlate || !!oneTimeLicencePlateError}
+              onPress={() => handleChoseVehicle(oneTimeLicencePlate)}
+            >
+              {t('actions.done')}
+            </Button>
+          ),
+        }}
+      />
       <ScreenContent>
-        <FlatList
-          data={vehicles}
-          keyExtractor={(vehicle) => vehicle.licencePlate}
-          ItemSeparatorComponent={() => <Divider dividerClassname="bg-transparent h-1" />}
-          renderItem={({ item: vehicleItem }) => (
-            <PressableStyled onPress={() => handleChoseVehicle(vehicleItem.licencePlate)}>
-              <VehicleRow
-                vehicle={vehicleItem}
-                selected={licencePlate === vehicleItem.licencePlate}
-              />
-            </PressableStyled>
-          )}
-        />
-        <Link asChild href="/vehicles/add-vehicle">
-          <Button variant="plain-dark" startIcon="add-circle-outline">
-            {t('addVehicle')}
-          </Button>
-        </Link>
+        <Field label={t('oneTimeUse')} errorMessage={oneTimeLicencePlateError}>
+          <TextInput
+            autoCapitalize="characters"
+            autoCorrect={false}
+            value={oneTimeLicencePlate}
+            onChangeText={handleLicencePlateChange}
+            hasError={!!oneTimeLicencePlateError}
+          />
+        </Field>
+
+        <View className="flex flex-row items-center">
+          <Divider dividerClassname="flex-grow" />
+          <Typography className="px-4">{t('chooseOtherOption')}</Typography>
+          <Divider dividerClassname="flex-grow" />
+        </View>
+
+        <View>
+          <Typography variant="default-bold" className="mb-2 grow">
```

Comment:
Instead of margins, pleas use `gap` on parent components.  Firstly, to match Figma, secondly, gap will never leave empty space when some of neighbouring elements is not present, margin will keep unwanted empty space. Therefore, gaps are more predictable and safe to use.

(Maybe another View is needed to wrap FlatList, but this is still more predictable.)
![image](https://github.com/bratislava/paas-mpa/assets/19418224/6d95034e-5ead-46b8-9343-ba77c264d3d4)